### PR TITLE
SCC: Promote sysctl annotations to fields

### PR DESCRIPTION
**Pull request by stlaz**
This PR promotes the experimental SCCs' sysctl-related annotations to fields.

- [x] promote sysctls from annotations to SCC fields
- [x] fix the sysctl-related tests disabled in the 1.11 rebase
- [x] extend the defaul `privileged` SCC to allow all sysctls

Trello: https://trello.com/c/WP50iqpB/222-add-forbiddensysctls-and-allowedunsafesysctls-options-to-sccs

### CODE REVIEW

File: pkg/oc/cli/describe/describer.go [hunk 1]
```diff
@@ -1879,6 +1879,13 @@ func describeSecurityContextConstraints(scc *securityapi.SecurityContextConstrai
 		fmt.Fprintf(out, "  Allowed Seccomp Profiles:\t%s\n", stringOrNone(strings.Join(scc.SeccompProfiles, ",")))
 		fmt.Fprintf(out, "  Allowed Volume Types:\t%s\n", fsTypeToString(scc.Volumes))
 		fmt.Fprintf(out, "  Allowed Flexvolumes:\t%s\n", flexVolumesToString(scc.AllowedFlexVolumes))
+
+		if len(scc.AllowedUnsafeSysctls) > 0 {
+			fmt.Fprintf(out, "  Allowed Unsafe Sysctls:\t%s\n", sysctlsToString(scc.AllowedUnsafeSysctls))
+		}
+		if len(scc.ForbiddenSysctls) > 0 {
+			fmt.Fprintf(out, "  Forbidden Sysctls:\t%s\n", sysctlsToString(scc.ForbiddenSysctls))
+		}
```
Comment (php-coder) [inline pkg/oc/cli/describe/describer.go]: Looks like we always show the fields. If their content is empty, we just represent it as `<all>` or `<none>`. So, could you remove these `if`-s then?

Comment (stlaz) [reply pkg/oc/cli/describe/describer.go]: Fixed :)

File: pkg/security/apis/security/validation/validation.go [hunk 2]
```diff
@@ -110,6 +111,98 @@ func ValidateSecurityContextConstraints(scc *securityapi.SecurityContextConstrai
...
+	forbiddenSysctlsPath := field.NewPath("forbiddenSysctls")
+	allErrs = append(allErrs, validateSCCSysctls(allowedUnsafeSysctlsPath, scc.AllowedUnsafeSysctls)...)
+	allErrs = append(allErrs, validateSCCSysctls(forbiddenSysctlsPath, scc.ForbiddenSysctls)...)
+	allErrs = append(allErrs, validatePodSecurityPolicySysctlListsDoNotOverlap(allowedUnsafeSysctlsPath, forbiddenSysctlsPath, scc.AllowedUnsafeSysctls, scc.ForbiddenSysctls)...)
+
+	return allErrs
+}
+
+const sysctlPatternSegmentFmt string = "([a-z0-9][-_a-z0-9]*)?[a-z0-9*]"
+const SysctlPatternFmt string = "(" + kapivalidation.SysctlSegmentFmt + "\\.)*" + sysctlPatternSegmentFmt
```
Comment (php-coder) [inline pkg/security/apis/security/validation/validation.go]: Could we make it non-pubic or something else depends on it?

Comment (stlaz) [reply pkg/security/apis/security/validation/validation.go]: I believe it should be alright to make it private, I'll change that.

**Comment by stlaz** [comment]
Apparently one of the dependencies from the bump is breaking the build:
```
# github.com/openshift/origin/vendor/github.com/spf13/viper
vendor/github.com/spf13/viper/viper.go:688:9: undefined: cast.ToInt32
```

**Comment by stlaz** [comment]
I suspect the tests that were previously failing are marked to be skipped now :thinking: 
@openshift/sig-security, @ingvagabund  this should be now good enough for a review

**Comment by simo5** [comment]
@deads2k can you approve ?

**Comment by enj** [comment]
/hold

https://github.com/openshift/api/pull/60 and any client-go changes need to merged first.

**Comment by liggitt** [comment]
openshift/api PR merged, can now drop the TMP commit and rerun update-deps

**Comment by liggitt** [comment]
/hold cancel
/approve
adding lgtm from https://github.com/openshift/origin/pull/20151#issuecomment-406561419

**Comment by liggitt** [comment]
travis failure is due to old commit checking, fixed in https://github.com/openshift/origin/pull/20381

CI is green otherwise, merging

**Comment by liggitt** [comment]
fixed bump commit title, re-tagging

**Pull request merged**